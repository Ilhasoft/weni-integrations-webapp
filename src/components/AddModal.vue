<template>
  <unnnic-modal
    ref="unnnic-add-modal"
    class="add-modal"
    :showModal="showAddModal"
    :text="$t('apps.details.actions.installed.title')"
    scheme="feedback-green"
    modal-icon="check-circle-1-1"
    @close="toggleModal"
  >
    <span slot="message" v-html="$t('apps.details.actions.installed.description')"></span>
    <unnnic-button
      ref="unnnic-add-modal-close-button"
      slot="options"
      type="terciary"
      @click="toggleModal"
      >{{ $t('general.Close') }}</unnnic-button
    >
    <unnnic-button
      ref="unnnic-add-modal-navigate-button"
      slot="options"
      type="primary"
      @click="navigateToMyApps"
    >
      {{ $t('apps.details.actions.installed.access_my_apps') }}
    </unnnic-button>
  </unnnic-modal>
</template>

<script>
  export default {
    name: 'AddModal',
    data() {
      return {
        showAddModal: false,
      };
    },
    methods: {
      toggleModal() {
        this.showAddModal = !this.showAddModal;
      },
      navigateToMyApps() {
        this.$router.replace('/apps/my');
      },
    },
  };
</script>

<style lang="scss" scoped>
  .add-modal {
    ::v-deep .unnnic-modal-container-background-body-close_icon {
      .unnnic-icon {
        display: none;
      }
    }
  }
</style>
