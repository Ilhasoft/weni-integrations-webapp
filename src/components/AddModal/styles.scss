.add-modal {
    ::v-deep .unnnic-modal-container-background-body-close_icon {
      .unnnic-icon {
        display: none;
      }
    }
  }