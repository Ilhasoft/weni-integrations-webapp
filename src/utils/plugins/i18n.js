import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
