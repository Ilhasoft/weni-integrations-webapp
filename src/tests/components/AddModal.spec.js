import { mount } from '@vue/test-utils';
import AddModal from '../../components/AddModal/index.vue';
import { describe, expect, it, beforeEach, vi } from 'vitest';
import i18n from '@/utils/plugins/i18n';
describe('AddModal', () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(AddModal, {
      global: {
        plugins: [i18n],
        mocks: {
          $router: {
            replace: vi.fn(),
          },
        },
      },
    });
    await wrapper.vm.$nextTick();
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('methods', () => {
    it('should change showModal', async () => {
      expect(wrapper.vm.showAddModal).toBe(false);
      await wrapper.vm.toggleModal();
      expect(wrapper.vm.showAddModal).toBe(true);
    });

    it('should navigate to myApps', async () => {
      await wrapper.vm.navigateToMyApps();
      expect(wrapper.vm.$router.replace).toHaveBeenCalledWith('/apps/my');
    });
  });
});
