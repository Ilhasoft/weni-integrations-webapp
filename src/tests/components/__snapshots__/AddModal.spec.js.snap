// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddModal > matches snapshot 1`] = `"<unnnic-modal data-v-abfb8ffe="" class="add-modal" showmodal="false" text="Installed Application!" scheme="feedback-green" modal-icon="check-circle-1-1"></unnnic-modal>"`;
